### ISSUE TITLE
@extends output missing selector when using same name for tag and attribute

### ISSUE THREAD

**Issue by jlowcs** [sass/libsass#2347]
### @extends output missing selector when using same name for tag and attribute

There is a strange issue when using @extend on a selector with a tag name and an attribute name with the same name: the second one is absent from the output.

input.scss
```scss
%baz2 {
    display: flex;
}
%baz3 {
    display: flex;
}

custom2, [custom2], .custom2 {
    @extend %baz2
}

[custom3], custom3, .custom3 {
    @extend %baz3
}
```

[libsass 3.5.0.beta.3][1]
```css
custom2, .custom2 {
  display: flex; }

[custom3], .custom3 {
  display: flex; }
```

Same result on http://sass.js.org/

version info:
```cmd
$ node-sass --version
node-sass       4.5.0   (Wrapper)       [JavaScript]
libsass         3.5.0.beta.2    (Sass Compiler) [C/C++]
```

Expected:
```css
custom2, [custom2], .custom2 {
  display: flex;
}

[custom3], custom3, .custom3 {
  display: flex;
}
```

Spec https://github.com/sass/sass-spec/pull/1091

[1]: http://libsass.ocbnet.ch/srcmap/#JWJhejIgewogICAgZGlzcGxheTogZmxleDsKfQolYmF6MyB7CiAgICBkaXNwbGF5OiBmbGV4Owp9CgpjdXN0b20yLCBbY3VzdG9tMl0sIC5jdXN0b20yIHsKICAgIEBleHRlbmQgJWJhejIKfQoKW2N1c3RvbTNdLCBjdXN0b20zLCAuY3VzdG9tMyB7CiAgICBAZXh0ZW5kICViYXozCn0=

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/109

**Comment by xzyfer** [comment]
#2350 only partially addressed this issue.

Before
```diff
-"custom2, [custom2], .custom2 {
+"custom2, .custom2 {
   display: flex;
 }
-[custom3], custom3, .custom3 {
+[custom3], .custom3 {
   display: flex;
 }
```
After
```diff
-"custom2, [custom2], .custom2 {
+"custom2, .custom2 {
   display: flex;
 }
 [custom3], custom3, .custom3 {
```

**Comment by mgreter** [comment]
Fixed by #2354
